app/src/routes/project/LayersPanel/index.tsx

```diff
@@ -4,10 +4,61 @@ import { EditorMode } from '@/lib/models';
 import { PinLeftIcon, PinRightIcon } from '@radix-ui/react-icons';
 import clsx from 'clsx';
 import { observer } from 'mobx-react-lite';
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
 import { useEditorEngine } from '..';
 import LayersTab from './LayersTab';
 import { capitalizeFirstLetter } from '/common/helpers';
+import { ReactComponentDescriptor } from '/electron/main/code/components';
+import { MainChannels } from '/common/constants';
+import { Button } from '@/components/ui/button';
+
+function ScanComponentsButton() {
+    const editorEngine = useEditorEngine();
+
+    const onClick = useCallback(async () => {
+        const path = (await window.api.invoke(MainChannels.PICK_COMPONENTS_DIRECTORY)) as
+            | string
+            | null;
+
+        if (path == null) {
+            return;
+        }
+
+        const components = (await window.api.invoke(
+            MainChannels.GET_COMPONENTS,
+            path,
+        )) as ReactComponentDescriptor[];
+        editorEngine.projectInfo.components = components;
+    }, [editorEngine]);
+
+    return (
+        <Button variant="outline" size="sm" className="" onClick={onClick}>
+            Connect to Project
+        </Button>
+    );
+}
+
+const ComponentsList = observer(({ components }: { components: ReactComponentDescriptor[] }) => {
```

Comment:
Style nite: I'd move this into its own component

Reply:
https://github.com/onlook-dev/onlook/pull/285/commits/9aad568f95b1e612fe70b707487567d7d6d4f13d